client/src/components/scheduler/PickTime.js

```diff
@@ -3,28 +3,34 @@ import { Grid, ListItem, makeStyles, Typography } from "@material-ui/core";
 import { Brightness1 } from "@material-ui/icons";
 import { format } from "date-fns";
 import PropTypes from "prop-types";
+import axios from "axios";
+import {getTimeSlots} from "../../utils/calendarUtils"
 
-const PickTime = (props) => {
+const PickTime = ({selectedDate, interval, availabilityTimes}) => {
   const classes = useStyles();
-  const selectedDate = props.selectedDate;
   const date = format(selectedDate, "EEEE, LLL do");
+  let isoDate=new Date(selectedDate);
+  isoDate.setHours(0,0,0,0);
+  isoDate=isoDate.toISOString();
+
 
   const [timeSlots, setTimeSlots] = useState([]);
+
+  //appointment length
   useEffect(() => {
     //fetch from backend
-
-    setTimeSlots([
-      "16:30",
-      "17:00",
-      "17:30",
-      "18:00",
-      "18:30",
-      "19:00",
-      "19:30",
-      "20:00",
-    ]);
+    axios.get(`/api/calendar/availability?day=${isoDate}`, {
+      withCredentials: true
+    }).then(res=>{
+      setTimeSlots(getTimeSlots(res.data.availability, interval, availabilityTimes))
+      }).catch(err=>console.log(err))
+    
   }, [date]);
```

Comment:
This isn't your line but I just noticed this. Since date isn't being set anywhere after its initialized is there any difference between this and just having this useEffect go onmount with `[]`?

Reply:
you're right! [] would work just as well